cdap-master/src/main/java/co/cask/cdap/data/tools/QueueConfigUpgrader.java

```diff
@@ -61,7 +61,13 @@ protected TableId getTableId() {
 
   @Nullable
   private QueueName fromRowKey(byte[] oldRowKey) {
-    QueueName queueName = QueueName.from(oldRowKey);
+    QueueName queueName;
+    try {
+      queueName = QueueName.from(oldRowKey);
```

Comment:
That's not a fool proof check. Actually for 2.6->2.8 upgrade, you can check if the version exists in the table attribute. Only 2.8 table has that attribute set.